### DISCUSSION TITLE
Unable to route traffic for tcp protocol using nginx-ingress-controller from Private Link Service(Azure)

### DISCUSSION THREAD

**Discussion by pandeyaws** [nginx/kubernetes-ingress#5957] [Q&A]
We have deployed nginx-ingress-controller using helm and override the tcp port using config map(overriding port for backend service) but its not able to listen on tcp port.
Ingress controller is able to listen on default port 443 and 80 but unable to listen on TCP port 5050(Custum port).
Following is the defination for configmap(TCP protocol):-->
apiVersion: v1
kind: ConfigMap
metadata:
  name: tcp-services
  namespace: nginx-ingress
data:
  5050: "default/xyz-svc:5050"

Ingress.yaml:
{{- if .Values.ingress.enabled }}
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: {{ .Values.deployment.name }}-ing
  namespace: {{ .Values.deployment.namespace }}
  annotations:
    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
spec:
  ingressClassName: nginx
  tls:
  - hosts:
    - {{ .Values.ingress.hosts }}
    secretName: {{ .Values.deployment.group }}-tls-secret
  rules:
  - host: {{ .Values.ingress.hosts }}
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: {{ .Values.ingress.backend.name }}
            port: 
              number: {{ .Values.ingress.backend.port }} 
  {{- if .Values.ingress.revpxy.enabled }} 
  - host: {{ .Values.ingress.hosts }}
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: {{ .Values.ingress.backend.name }}
            port: 
              number: {{ .Values.ingress.revpxy.backend.port }} 
            
{{- end }}
{{- end }}

**Discription**: 
Based on ingress rule, we are routing traffic based on ingress host name for example if the hostname = xyz then it should route request to xyz-svc on port 5050 .
And if the hostname = ABC then it should routhe request to ABC-svc on port 5050 .
 but we are not able to route request based on host name && getting connection timeout .

We are looking some support/suggestion on how to proceed further.

**Comment by haywoodsh** [comment]
Hi @pandeyaws, can you confirm you are indeed using the NGINX Ingress Controller in this repo, and if so, can you please let me know the version you were using? I noticed that you were using the annotation `nginx.ingress.kubernetes.io/backend-protocol` which is not in our list of supported annotations. https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-annotations/
Additionally, as this is not a Helm chart we provided, to ensure that the issue is not related to Helm, could you please provide the generated ingress specification by using the command `kubectl describe ingress <name> -n <namespace>`?
Generally, if you need to open ports other than 80 and 443 on a cloud platform, you might want to refer to the platform's documentation for deploying a vendor-specific LoadBalancer, and ensure that the security and network policies are configured correctly to allow traffic on the custom port.

**Comment by vepatel** [comment]
@pandeyaws is this resolved? can we close this discussion?
